src/php/lib/Grpc/BaseStub.php

```diff
@@ -72,11 +72,21 @@ public function __construct($hostname, $opts, Channel $channel = null)
                                  'ChannelCredentials::create methods');
         }
         if ($channel) {
-            if (!is_a($channel, 'Grpc\Channel')) {
-                throw new \Exception('The channel argument is not a'.
-                                     'Channel object');
+            if (!is_a($channel, 'Grpc\Channel') &&
+                !is_a($channel, 'Grpc\InterceptorChannel')) {
+                throw new \Exception('The channel argument is not a Channel object '.
+                    'or an InterceptorChannel object created by '.
+                    'InterceptorHelper::intercept($channel, Interceptor $interceptor)');
             }
             $this->channel = $channel;
+            while(is_a($channel, 'Grpc\InterceptorChannel')) {
```

Comment:
I don't think this whole check (the while loop and the following if statement) is necessary. The `InterceptorChannel` construction process ensures this invariant is true.

Reply:
InterceptorChannel construction process doesn't check the deepest channel which can be anything because next pointer can points to everything. I'll add a check it during construction.

Reply:
But it does not need to! At most, it just needs to verify that the `$channel` it gets is a valid `Channel` or `InterceptorChannel`. Inductively, that ensures your entire `InterceptorChannel` construction is solid. I think looping through the channel linked-list is breaking encapsulation and not a good idea in general.